tools/inference_engine/src/impl/model.h

```diff
@@ -246,6 +249,47 @@ class Activation : public INode
     GpuKernel::Ptr kernel_;
 };
 
+class DAG {
```

Comment:
For now it's OK,
To investigate/think of:
a) does DAG should be owner of the nodes, so user can release them
b) does DAG should owner of the graph, so calls like topological_sort() are not exposed, but called after graph building phase is finished
b) DAG registration of optimization passes and utility codes for it (like removal of node which is fused with predecessor)